lib/apis/v3/felixconfig.go

```diff
@@ -113,22 +110,22 @@ type FelixConfigurationSpec struct {
 	// “INPUT” chain; Calico will insert its rules at the top of that chain, then “RETURN” packets to the “INPUT” chain
 	// once it has completed processing workload endpoint egress policy. Use ACCEPT to unconditionally accept packets
 	// from workloads after processing workload endpoint egress policy. [Default: DROP]
-	DefaultEndpointToHostAction string `json:"defaultEndpointToHostAction,omitempty"`
-	IptablesFilterAllowAction   string `json:"iptablesFilterAllowAction,omitempty"`
-	IptablesMangleAllowAction   string `json:"iptablesMangleAllowAction,omitempty"`
+	DefaultEndpointToHostAction string `json:"defaultEndpointToHostAction,omitempty" validate:"omitempty,dropacceptreturn"`
+	IptablesFilterAllowAction   string `json:"iptablesFilterAllowAction,omitempty" validate:"omitempty,acceptreturn"`
+	IptablesMangleAllowAction   string `json:"iptablesMangleAllowAction,omitempty" validate:"omitempty,acceptreturn"`
 	// LogPrefix is the log prefix that Felix uses when rendering LOG rules. [Default: calico-packet]
 	LogPrefix string `json:"logPrefix,omitempty"`
 
 	// LogFilePath is the full path to the Felix log. Set to none to disable file logging. [Default: /var/log/calico/felix.log]
 	LogFilePath string `json:"logFilePath,omitempty"`
 
 	// LogSeverityFile is the log severity above which logs are sent to the log file. [Default: INFO]
-	LogSeverityFile string `json:"logSeverityFile,omitempty"`
+	LogSeverityFile string `json:"logSeverityFile,omitempty" validate:"omitempty,felixLogLevel"`
 	// LogSeverityScreen is the log severity above which logs are sent to the stdout. [Default: INFO]
-	LogSeverityScreen string `json:"logSeverityScreen,omitempty"`
+	LogSeverityScreen string `json:"logSeverityScreen,omitempty" validate:"omitempty,felixLogLevel"`
 	// LogSeveritySys is the log severity above which logs are sent to the syslog. Set to NONE for no logging to syslog.
 	// [Default: INFO]
-	LogSeveritySys string `json:"logSeveritySys,omitempty"`
+	LogSeveritySys string `json:"logSeveritySys,omitempty" validate:"omitempty,felixLogLevel"`
 
 	IPIPEnabled *bool `json:"ipipEnabled,omitempty" confignamev1:"IpInIpEnabled"`
 	// IPIPMTU is the MTU to set on the tunnel device. See Configuring MTU [Default: 1440]
```

Comment:
`ProtoPort` needs some validation tags
`     |`
`     |`
`     V`

Reply:
added validation + tests for ProtoPort to ensure Protocol is either `TCP` or `UDP`.

Reply:
ok - i guess I was expecting that you'd just add a regex validator to the Protocol field.  No worries though - as long as it's validated.